app/models/bottleneck_event.rb

```diff
@@ -101,11 +101,10 @@ def self.calculate_future_trend_to_limit(obj, options)
     result
   end
 
-  def self.event_where_clause(obj)
-    ids_hash = child_types_and_ids(obj)
-    result = ["(resource_type = '#{obj.class.base_class.name}' AND resource_id = #{obj.id})"]
-    ids_hash.each { |k, v| result.push("(resource_type = '#{k}' AND resource_id in (#{v.join(",")}))") }
-    result.join(" OR ")
+  def self.event_where_clause(scope, obj)
+    child_types_and_ids(obj).each do |k, v|
+      scope = scope.or(resource_type => obj.class.base_class.name, "resource_id" => obj.id)
+    end
   end
```

Comment:
`event_where_clause ` returns a hash now. Shouldn't it return a scope? Also, `k` and `v` parameters are not used at all. Are there any tests that cover this method?